main.tf

```diff
@@ -17,11 +15,11 @@ resource "null_resource" "provisioner" {
 
   triggers {
     signature = "${data.archive_file.default.output_md5}"
-    command   = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", var.arguments)} -e ${join(" -e ", var.envs)} ${var.playbook}"
+    command   = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", compact(var.arguments))} ${length(compact(var.envs)) > 0 ? "-e" : ""} ${length(compact(var.envs)) > 0 ? join(" -e ", var.envs) : ""} ${var.playbook}"
```

Comment:
~shouldn't there be spaces on either side of the `-e` like we do in the `join`?~

Ah, I see. there are spaces outside of the interpolation.

Reply:
`${length(compact(var.envs)) > 0 ? join(" -e ", var.envs) : ""}`
should be
`${join(" -e ", compact(var.envs))}`